x-pack/auditbeat/module/system/socket/kprobes.go

```diff
@@ -8,134 +8,19 @@ package socket
-		pdef = d(pdef)
-	}
-	if pdef.Decoder == nil {
-		return format, decoder, errors.New("nil decoder in probe definition")
-	}
-	if err = p.traceFS.AddKProbe(pdef.Probe); err != nil {
-		return format, decoder, errors.Wrapf(err, "failed installing probe '%s'", pdef.Probe.String())
-	}
-	p.installed = append(p.installed, pdef.Probe)
-	if format, err = p.traceFS.LoadProbeFormat(pdef.Probe); err != nil {
-		return format, decoder, errors.Wrap(err, "failed to load probe format")
-	}
-	if decoder, err = pdef.Decoder(format); err != nil {
-		return format, decoder, errors.Wrap(err, "failed to create decoder")
-	}
-	return
-}
-
-// UninstallInstalled uninstalls the probes installed by Install.
-func (p *probeInstaller) UninstallInstalled() error {
-	var errs multierror.Errors
-	for _, probe := range p.installed {
-		if err := p.traceFS.RemoveKProbe(probe); err != nil {
-			errs = append(errs, err)
-		}
-	}
-	p.installed = nil
-	return errs.Err()
-}
-
-// UninstallIf uninstalls all probes in the system that met the condition.
-func (p *probeInstaller) UninstallIf(condition helper.ProbeCondition) error {
-	kprobes, err := p.traceFS.ListKProbes()
-	if err != nil {
-		return errors.Wrap(err, "failed to list installed kprobes")
-	}
-	var errs multierror.Errors
-	for _, probe := range kprobes {
-		if condition(probe) {
-			if err := p.traceFS.RemoveKProbe(probe); err != nil {
-				errs = append(errs, errors.Wrapf(err, "unable to remove kprobe '%s'", probe.String()))
-			}
-		}
-	}
-	return errs.Err()
-}
-
-// WithGroup sets a custom group to probes before they are installed.
-func WithGroup(name string) ProbeTransform {
-	return func(probe helper.ProbeDef) helper.ProbeDef {
-		probe.Probe.Group = name
-		return probe
-	}
-}
-
-// WithTemplates expands templates in probes before they are installed.
-func WithTemplates(vars common.MapStr) ProbeTransform {
-	return func(probe helper.ProbeDef) helper.ProbeDef {
-		return probe.ApplyTemplate(vars)
-	}
-}
-
-// WithNoOp is a no-op transform.
-func WithNoOp() ProbeTransform {
-	return func(def helper.ProbeDef) helper.ProbeDef {
-		return def
-	}
-}
-
-// WithFilterPort is used for filtering port 22 traffic when debugging over
-// an SSH connection. Otherwise there is a feedback loop when tracing events
-// printed on the terminal are transmitted over SSH, which causes more tracing
-// events.
-func WithFilterPort(portnum uint16) ProbeTransform {
-	var buf [2]byte
-	tracing.MachineEndian.PutUint16(buf[:], portnum)
-	filter := fmt.Sprintf("lport!=0x%02x%02x", buf[0], buf[1])
-	return func(probe helper.ProbeDef) helper.ProbeDef {
-		if strings.Contains(probe.Probe.Fetchargs, "lport=") {
-			if probe.Probe.Filter == "" {
-				probe.Probe.Filter = filter
-			} else {
-				probe.Probe.Filter = fmt.Sprintf("%s && (%s)",
-					filter, probe.Probe.Filter)
-			}
-		}
-		return probe
-	}
-}
-
 // KProbes shared with IPv4 and IPv6.
-var sharedKProbes = []helper.ProbeDef{
+var sharedKProbes = []kprobes.ProbeDef{
```

Comment:
Though not necessary in this commit, it'd probably be worth splitting these into their own files similar to the other `guess`-based files, just for easier management.